javascript/main.js

```diff
@@ -0,0 +1,205 @@
+const newShoppingListItem = (item,price) => ({item, price,})
+const addToShoppingList = (item,list=[]) => list.concat([item]);
+const removeFromShoppingList = (list=[]) => list.slice(0, -1);
+const removeFirstItem = (list=[]) => list.slice(1);
+
+const removeNthItem = (i,list = []) => {
+	if (i<list.length && i>=0 && Number.isInteger(i) == true) {
+		list.splice(i, 1);
+	} else {
+		throw new Error('error')
+	}
+	return list;
+};
+
+//this is the controller for the messaging div
+const headsUp = document.querySelector('.js-warning');
+
+//Redraw the js-data div with the new list
+const render = () => {
+	let htmlStr = '<table class="table list-group">';
+	htmlStr += `
+	<thead>
+	<tr>
+	<th scope="col">#</th>
+	<th scope="col">Item</th>
+	<th scope="col">Price</th>
+	</tr>
+	</thead>
+	<tbody>
+	`;
+	for(let i = 0; i < shoppingList.length; i++) {
+		n = i + 1;
+		htmlStr += `
+		<tr class="list-group-item">
+		<th scope="row">${n}</th>
+		<td class="rowName">${shoppingList[i].item}</td>
+		<td>${shoppingList[i].price}</td>
+		<td><button class="btn removeBtn">X</button></td>
+		`
+	}
+	drawTotal = totalCost();
+	htmlStr += `
+	</tbody>
+	</table>
+	<div class="card">
+	TOTAL COST    $ ${drawTotal}
+	</div>`
+	document.querySelector('.js-data').innerHTML = htmlStr;
+	const budgetVal = Number(budget.value);
+	if (budgetVal === 0 || drawTotal <= budgetVal) {
+		headsUp.style.color = "black";
+		headsUp.style.textTransform = "none";
+		headsUp.innerHTML = "Please fill out the list.";
+	}
+	else {
+		if (drawTotal > budgetVal) {
+			headsUp.style.color = "red";
+			headsUp.style.textTransform = "uppercase";
+			headsUp.innerHTML = "You have gone over your budget";
+		}
+	}
+}
+
+//Variable to hold the actual list
+let shoppingList = [];
+
+//budget controller
+const budget = document.querySelector('.js-budget');
+budget.addEventListener('keydown', e => {
+	if (e.keyCode === 13) {
+		e.preventDefault();
+		console.log("budget refresh")
+		render();
+	}
+});
+
+
+//Var for the contents of the input line
+const shoppingInput = document.querySelector('.js-shopping-list-input');
+
+//wait for enter press on the name of the item
+shoppingInput.addEventListener('keydown', e => {
+	if (e.keyCode === 13) {
+		e.preventDefault();
+		mainItemAdd();
+	}
+});
+
+////Var for the contents of the price line
+const priceInput = document.querySelector('.js-shopping-list-price');
+
+//wait for enter press on the price of the item
+priceInput.addEventListener('keydown', e => 
+{
+	//exhaustive testing to make sure there is a number in the price field
+	let wrongKey = false;
+	const nonNumberCodes = [186, 187, 188, 189, 191, 192, 219, 220, 221, 220];
```

Comment:
This is ugly but you're absolutely correct that this is the way to do it. However, there are more optimal ways of doing this. You can take a look at [this](https://stackoverflow.com/questions/31852843/checking-regex-expression-on-keypress) which gets you started on the idea, but it's not fully formed for just selecting numbers. I don't expect you to change your homework; this is just for your knowledge and enrichment.